var/spack/repos/builtin/packages/vecgeom/package.py

```diff
@@ -174,10 +173,16 @@ class Vecgeom(CMakePackage, CudaPackage):
         when="@1.1.18 +cuda ^cuda@:11.4",
     )
 
-    for std in _cxxstd_values:
-        depends_on("geant4 cxxstd=" + std, when="+geant4 cxxstd=" + std)
-        depends_on("root cxxstd=" + std, when="+root cxxstd=" + std)
-        depends_on("xerces-c cxxstd=" + std, when="+gdml cxxstd=" + std)
+    for _cxxstd in _cxxstd_values:
+        for _v in (
+            _cxxstd
+            if isinstance(_cxxstd, _ConditionalVariantValues)
+            else [Value(_cxxstd, when="")]
+        ):
```

Comment:
Or maybe the above suggestion could be a helper function in `spack.variant`?

Reply:
This is an interesting suggestion. I am not sure how it would work. Ideally it would plug into the variant itself, so we would do something like an iterable `self.variants["cxxstd"].values` but that would only be accessible in actual member functions when `self` exists, not in these patterns. Here we would have to rely on operating on the user-defined iterable that is provided to values in the variant definition. In C++ I'd call it a static function exposed to the users, which would not be considered a recommended solution in C++...

Reply:
Fair enough. Maybe just keep it as a local function (or `@staticmethod`).

Reply:
The `@staticmethod` approach fails on python 3.6 in the rhel8 pipelines, https://github.com/spack/spack/actions/runs/6364026233/job/17280024733?pr=39785, so I'm going to go back to a local function.

Reply:
Local generator functions introduced in both geant4 and vecgeom.